ddtrace/contrib/starlette/patch.py

```diff
@@ -150,3 +160,20 @@ def traced_handler(wrapped, instance, args, kwargs):
         raise trace_utils.InterruptException("starlette")
 
     return wrapped(*args, **kwargs)
+
+
+@with_traced_module
+def _trace_background_tasks(module, pin, wrapped, instance, args, kwargs):
+    task = get_argument_value(args, kwargs, 0, "func")
+    current_span = pin.tracer.current_span()
+
+    async def traced_task(*args, **kwargs):
+        with pin.tracer.start_span(
+            f"{module.__name__}.background_task", resource=task.__name__, child_of=None, activate=True
+        ) as span:
+            if current_span:
+                span.link_span(current_span.context)
+            await task(*args, **kwargs)
```

Comment:
The task isn't always await-able. Fastapi checks if this function is async and we should do the same: https://github.com/tiangolo/starlette/blob/654a65b6495862bbbffaaad8e9b04c8d50653451/starlette/background.py#L18

```suggestion
           if inspect. isawaitable(task)
               await task(*args, **kwargs)
           else:
               .... # make function awaitable? 
               
```

Reply:
This should address this test failure: https://github.com/DataDog/dd-trace-py/actions/runs/7645610961/job/20832607414?pr=7618

Reply:
OK I'll try some things with what you have started here 👍